tests/Unit/IO/Test_H5.cpp

```diff
@@ -267,6 +269,10 @@ SPECTRE_TEST_CASE("Unit.IO.H5.Dat", "[Unit][IO][H5]") {
 
   h5::H5File<h5::AccessType::ReadWrite> my_file(h5_file_name);
   my_file.insert<h5::Dat>("/L2_errors", legend, version_number);
+
+  // Check that the Dat file is found to be a subgroup of the file
+  CHECK((my_file.groups())[0] == "L2_errors.dat");
```

Comment:
since we can't rely on the order that HDF5 gives us groups in could you do this check as
```cpp
const auto groups_in_file = my_file.groups();
CHECK(alg::find(groups_in_file, std::string{"L2_errors.dat"}) != groups_in_file.end());
```
The function `alg::find` is in `Utilities/Algorithm.hpp` :)